### DISCUSSION TITLE
Is sampler1D datatype supported as output in node?

### DISCUSSION THREAD

**Discussion by HSDSZ** [bnpr/Malt#184] [Q&A] [answered]
If I am right, sampler1D datatype is that slide bar. 
Is sampler1D datatype supported as output in node or via inline code?
So I can  use one sampler1D to control both "get_diffuse_gradient" and "sampler1D_sample" .
![image](https://user-images.githubusercontent.com/36077492/143615077-8b9a77f6-e44e-4d82-b952-ec7c56ae68d4.png)

**Comment by pragma37** [comment, marked as answer]
Yep!
![imagen](https://user-images.githubusercontent.com/36610999/143615603-cd802c93-20e6-41f0-bef4-d17be4ff9fa6.png)
This is the node source code:
```glsl
sampler1D sampler1D_prop(sampler1D s) { return s; }
```
Should work the same for sampler2D.

**Reply by HSDSZ** [reply]
Ok, got it
Then the inline code version is 
![image](https://user-images.githubusercontent.com/36077492/143616159-7525c41b-ec4a-42a5-a11a-05008de854bd.png)

**Reply by pragma37** [reply]
Yes, that should work as well.
